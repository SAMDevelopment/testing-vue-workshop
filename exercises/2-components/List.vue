<template>
  <ul v-if="items">
    <li 
      v-for="(item, i) in items"
      :key="`item-${i}`"
    >
      {{ item }}
    </li>
  </ul>
  <span v-else>No items</span>
</template>

<script>
export default {
  props: {
    items: {
      type: Array,
      default: undefined
    }
  }
}
</script>
