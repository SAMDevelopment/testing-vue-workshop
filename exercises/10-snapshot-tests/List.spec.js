import { shallowMount } from '@vue/test-utils'
import List from './List.vue'

test('renders list of items correctly', () => {
  const wrapper = shallowMount(List, {
    propsData: {
      items: (new Array(5)).fill().map((value, index) => index)
    }
  })
  expect(wrapper.element).toMatchSnapshot()
})
