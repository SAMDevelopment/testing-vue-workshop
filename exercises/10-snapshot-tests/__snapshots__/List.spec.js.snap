// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders list of items correctly 1`] = `
<ul>
  <li>
    
    0
  
  </li>
  <li>
    
    1
  
  </li>
  <li>
    
    2
  
  </li>
  <li>
    
    3
  
  </li>
  <li>
    
    4
  
  </li>
</ul>
`;
